homeassistant/components/vesync/switch.py

```diff
@@ -112,9 +132,9 @@ def is_on(self) -> bool | None:
     def turn_off(self, **kwargs: Any) -> None:
         """Turn the entity off."""
         if self.entity_description.off_fn(self.device):
-            self.schedule_update_ha_state()
+            self.schedule_update_ha_state(force_refresh=True)
```

Comment:
Why do we need to force_refresh ?

The changed lines in turn_on/turn_off are getting unit tested.

Reply:
Was finding without it the toggle was waiting for next refresh.  Maybe another internal state thing in the library?

Reply:
To simplify and I will remove this for now.

Reply:
I looked into the library and the integration and now I understand it. 

If you look at `set_display` in vesyncfan.py, you will see that it invokes a web request but never updates the device state dictionary itself. So the device at HA end continues to be in previous state, the state only updates on next refresh. A forced refresh would update the state sooner than the next scheduled update.

Reply:
Yes that makes sense.   In my other integrations I took the approach of not trusting the physical device did indeed change so rather always check the raw data, however that was cars to less reliable.

Reply:
I think forced refresh should be done on a case by case basis otherwise we lose the benefit of scheduled refreshes. This sort of a discrepancy in the library. Methods such as turn_on/turn_off do update the device state.

Reply:
Your initial approach to force refresh feels valid here.

Reply:
We need to update the documentation for these new settings.

Reply:
Documentation added.  I am thinking the force refresh items should live in this: https://github.com/home-assistant/core/pull/135958 since that is the larger fix for state issues.